lib/Controller/RoomController.php

```diff
@@ -456,6 +467,60 @@ protected function createGroupRoom(string $targetGroupName): DataResponse {
 		return new DataResponse($this->formatRoom($room, $room->getParticipant($currentUser->getUID())), Http::STATUS_CREATED);
 	}
 
+	/**
+	 * Initiates a group video call from the selected circle
+	 *
+	 * @NoAdminRequired
+	 *
+	 * @param string $targetCircleId
+	 * @return DataResponse
+	 */
+	protected function createCircleRoom(string $targetCircleId): DataResponse {
+		if (!$this->appManager->isEnabledForUser('circles')) {
+			return new DataResponse([], Http::STATUS_BAD_REQUEST);
+		}
+
+		/** @var Circles $circlesApi */
+		try {
+			$circle = Circles::detailsCircle($targetCircleId);
+		} catch (\Exception $e) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+		$currentUser = $this->userManager->get($this->userId);
+
+		if (!$currentUser instanceof IUser) {
+			return new DataResponse([], Http::STATUS_NOT_FOUND);
+		}
+
+		// Create the room
+		$room = $this->manager->createGroupRoom($circle->getName());
+		$room->addUsers([
+			'userId' => $currentUser->getUID(),
+			'participantType' => Participant::OWNER,
+		]);
+
+		$participants = [];
+		foreach ($circle->getMembers() as $member) {
+			/** @var Member $member */
+			if ($member->getType() === Member::TYPE_USER || $member->getUserId() === '') {
```

Comment:
I would add a check on the Level: $member->isLevel(MEMBER::LEVEL_MEMBER) as in current state, non-yet members (invited or requesting for access) will be invited to the room.
@daita